metadata_layer.py

```diff
@@ -721,7 +721,7 @@ def mirror_link(self, model_object, data_source, link, link_obj, policy):
             )
             return
 
-        mirror = policy.mirror
+        mirror = policy.mirror[mirror_type]
```

Comment:
You need to handle the case where `mirror_type` doesn't show up in the dictionary.